src/dialogs/create-file.tsx

```diff
@@ -65,22 +66,23 @@ async function create_file(filename: string, content?: string, owner?: string |
     }
 
     if (content) {
-        // We need to obtain the tag to retain file ownership
-        // Can't do this async because we can't get the tag via await
-        await cockpit.file(filename, { superuser: "try" }).read()
-                .then(((async (_content: string, tag: string) => {
-                    try {
-                        await cockpit.file(filename, { superuser: "try" }).replace(content, tag);
-                    } catch (err) {
-                        console.warn("Cannot set initial file text", err);
-                        try {
-                            await cockpit.spawn(["rm", filename], { superuser: "require" });
-                        } catch (err) {
-                            console.warn(`Failed to cleanup ${filename}`, err);
-                        }
-                        throw err;
-                    }
-                }) as any /* eslint-disable-line @typescript-eslint/no-explicit-any */));
+        try {
+            // We need to obtain the tag to retain file ownership
+            const { tag } = await fsinfo(filename, ['tag'], { superuser: "try" });
+            try {
+                await cockpit.file(filename, { superuser: "try" }).replace(content, tag);
+            } catch (err) {
+                console.warn("Cannot set initial file text", err);
+                try {
+                    await cockpit.spawn(["rm", filename], { superuser: "require" });
```

Comment:
I know this is existing behaviour, but why should we try to delete the file if we failed to create it?  What if the previous file is still there?  This seems kinda random...

Reply:
Good question, for the chown case it makes sense to remove it, as if that fails we don't want to keep that around.

But if we fail to set the contents, should we remove it? It might have been modified so that doesn't sound great.

Reply:
I mean, if the fchown fails inside of the channel implementation that we won't override the original file to begin with, right?  So 'rm' here will only ever succeed in deleting a previous file....